cloudferrylib/os/compute/nova_compute.py

```diff
@@ -658,3 +669,19 @@ def get_free_disk(self):
 
     def delete_vm_by_id(self, vm_id):
         self.nova_client.servers.delete(vm_id)
+
+
+def down_hosts(nova_client_):
+    services = nova_client_.services.list()
+    return set(map(attrgetter('host'),
+                   filter(lambda h: h.state != 'up', services)))
```

Comment:
Seriously, there's no need to attach this method to any class, it doesn't need to be part of any object, which makes it more reusable